scripts/jest/ts-preprocessor.js

```diff
@@ -71,18 +71,17 @@ function compile(content, contentFilename) {
     getNewLine: function() {
       return '\n';
     },
+    fileExists: function (filename) {
+      return ts.sys.fileExists(filename);
+    },
   };
   var program = ts.createProgram([
     'lib.d.ts',
     'jest.d.ts',
     contentFilename,
   ], tsOptions, compilerHost);
-  var errors = program.getDiagnostics();
-  if (!errors.length) {
-    var checker = program.getTypeChecker(true);
-    errors = checker.getDiagnostics();
-    checker.emitFiles();
-  }
+  var emitResult = program.emit();
+  var errors = ts.getPreEmitDiagnostics(program).concat(emitResult.diagnostics);
```

Comment:
Can you double (triple?) check this by putting some invalid type info into jest.d.ts? I want to make sure tests still fail in that case. I remember having some issues in the past when I tried to upgrade TS to 1.5.